client/jetpack-connect/plans.jsx

```diff
@@ -115,8 +129,14 @@ class Plans extends Component {
 		}
 	}
 
-	redirect( path ) {
-		page.redirect( path + this.props.selectedSiteSlug );
+	redirect( path, args ) {
+		let redirectTo = path + this.props.selectedSiteSlug;
+
+		if ( args ) {
+			redirectTo = addQueryArgs( args, redirectTo );
+		}
```

Comment:
Minor: We could drop any branching and mutation:

```js
redirect( path, args = {} ) {
  const redirectTo = addQueryArgs( args, path + this.props.selectedSiteSlug );
  // …
}
```

Take it or leave it 🙂

Reply:
I think I'll skip this suggestion, because with this version in the case of empty `args` we'll unnecessarily perform some unnecessary object assignments and URL parsing, and with the check this PR is suggesting we're avoiding that.